packages/lit-auth-client/src/lib/utils.ts

```diff
@@ -194,3 +197,131 @@ export function getRPIdFromOrigin(origin: string) {
   // remove port with regex
   return newOrigin.replace(/:\d+$/, '');
 }
+
+/**
+ * Derive unique identifier from authentication material produced by auth providers
+ *
+ * @param {AuthMethod} authMethod - Auth method object from provider
+ *
+ * @returns {Promise<string>} - Auth method id
+ */
+export async function getAuthMethodId(authMethod: AuthMethod): Promise<string> {
```

Comment:
if this is being exposed for using in the contracts we will want to allow non mapped auth methods for users custom auth types.